inputfiles/idl/XMLHttpRequest.commentmap.json

```diff
@@ -1,15 +1,15 @@
-    "xmlhttprequest-upload": "Returns the associated XMLHttpRequestUpload object. It can be used to gather transmission information when data is\ntransferred to a server.",
-    "xmlhttprequest-send": "Initiates the request. The body argument provides the request body, if any,\nand is ignored if the request method is GET or HEAD.\nThrows an \"InvalidStateError\" DOMException if either state is not opened or the send() flag is set.",
+    "xmlhttprequest-open": "Sets the request method, request URL, and synchronous flag.\nThrows a \"SyntaxError\" DOMException if either method is not a valid HTTP method or url cannot be parsed.\nThrows a \"SecurityError\" DOMException if method is a case-insensitive match for `CONNECT`, `TRACE`, or `TRACK`.\nThrows an \"InvalidAccessError\" DOMException if async is false, current global object is a Window object, and the timeout attribute is not zero or the responseType attribute is not the empty string.",
+    "xmlhttprequest-setrequestheader": "Combines a header in author request headers.\nThrows an \"InvalidStateError\" DOMException if either state is not opened or the send() flag is set.\nThrows a \"SyntaxError\" DOMException if name is not a header name or if value is not a header value.",
+    "xmlhttprequest-timeout": "Can be set to a time in milliseconds. When set to a non-zero value will cause fetching to terminate after the given time has passed. When the time has passed, the request has not yet completed, and the synchronous flag is unset, a timeout event will then be dispatched, or a \"TimeoutError\" DOMException will be thrown otherwise (for the send() method).\nWhen set: throws an \"InvalidAccessError\" DOMException if the synchronous flag is set and current global object is a Window object.",
+    "xmlhttprequest-withcredentials": "True when credentials are to be included in a cross-origin request. False when they are to be excluded in a cross-origin request and when cookies are to be ignored in its response. Initially false.\nWhen set: throws an \"InvalidStateError\" DOMException if state is not unsent or opened, or if the send() flag is set.",
+    "xmlhttprequest-upload": "Returns the associated XMLHttpRequestUpload object. It can be used to gather transmission information when data is transferred to a server.",
+    "xmlhttprequest-send": "Initiates the request. The body argument provides the request body, if any, and is ignored if the request method is GET or HEAD.\nThrows an \"InvalidStateError\" DOMException if either state is not opened or the send() flag is set.",
     "xmlhttprequest-abort": "Cancels any network activity.",
-    "xmlhttprequest-overridemimetype": "Acts as if the `Content-Type` header value for response is mime.\n(It does not actually change the header though.)\nThrows an \"InvalidStateError\" DOMException if state is loading or done.",
```

Comment:
The newlines make reading documentation a lot easier for long comments. I tested VS, VS Code and emacs.

VS and emacs treat newline as newline.
VS Code treats two-spaces-before-newline as newline -- that is, it's interpreting the comment as markdown.

I think the right thing to do is to convert the original newlines into '  \n' so that both approaches will look good.

Reply:
I did the `innerText` way because the original newlines were for spec authors who edit raw HTML, which tends to look arbitrary when extracted as comments. In the worst case a full line only included a single word, which was perfectly reasonable in raw HTML because it was wrapped by an anchor element.

That said, I fixed a bug that `<p>` didn't insert double newlines. If it still doesn't look good, then I can skip and defer the newline formatting part here.

Reply:
Hmm. I guess it's OK, but maybe we need to have our own wrapping algorithm eventually? The editors I tried don't wrap on their own [1], which becomes pretty hard to read.

[1] VS wraps to the width of the screen, emacs wraps to the width of the window. These are often nearly the same.

Reply:
>but maybe we need to have our own wrapping algorithm eventually?

Hmm, I'll try it.